queries/soql/highlights.scm

```diff
@@ -0,0 +1,142 @@
+; https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#semantic-token-classification
+(field_identifier
+  (identifier) @property)
+
+(field_identifier
+  (dotted_identifier
+    (identifier) @property))
+
+(type_of_clause
+  (identifier) @property)
+
+(when_expression
+  (identifier) @type)
+
+(when_expression
+  (field_list
+    (identifier) @property))
+
+(when_expression
+  (field_list
+    (dotted_identifier
+      (identifier) @property )))
+
+(else_expression
+  (field_list
+    (identifier) @property ))
+
+(else_expression
+  (field_list
+    (dotted_identifier
+      (identifier) @property )))
+
+(alias_expression
+  (identifier) @label)
+
+(storage_identifier) @type
+(function_name) @function
+(date_literal) @variable.readonly.defaultLibrary
+
+[
+  ","
+  "."
+  ":"
+  "?"
+  "("
+  ")"
+] @punctuation.delimiter
+
+[
+  "AND"
+  "OR"
+  "NOT"
+  "="
+  "!="
+  "LIKE"
+  "NOT_IN"
+  "INCLUDES"
+  "EXCLUDES"
+] @operator
+(value_comparison_operator "<" @operator)
+"<=" @operator
+(value_comparison_operator ">" @operator)
+">=" @operator
+ @operator
+(set_comparison_operator "IN" @operator)
+
+(int) @number
+(decimal) @number
+(currency_literal) @number
+(string_literal) @string
+(date) @variable.readonly
+(date_time) @variable.readonly
```

Comment:
How do these differ from `(date_literal)`?

Reply:
not sure here. @aheber comments?

Reply:
`date_literal` are macros for relative moments: `TODAY`, `YESTERDAY`, `LAST_YEAR`
https://github.com/aheber/tree-sitter-sfapex/blob/e63bcdcc26ae808b3fe79dfb8fa61bebdb95bda4/common/soql-grammar.js#L328

`date` is a raw date string `YYYY-MM-DD`
https://github.com/aheber/tree-sitter-sfapex/blob/e63bcdcc26ae808b3fe79dfb8fa61bebdb95bda4/common/soql-grammar.js#L450

`date_time` is a raw datetime string `YYYY-MM-DDThh:mm:ss.mmmmZ`
https://github.com/aheber/tree-sitter-sfapex/blob/e63bcdcc26ae808b3fe79dfb8fa61bebdb95bda4/common/soql-grammar.js#L452

I tagged them differently because the macro format receives a moment-relative value when processed, it doesn't have a constant inherent value like the `date` and `date_time` expressions do.

Reply:
In this case, I guess no change is needed? @ObserverOfTime